server/util/grpc_client/grpc_client.go

```diff
@@ -133,6 +134,61 @@ func (p *ClientConnPool) NewStream(ctx context.Context, desc *grpc.StreamDesc, m
 	return p.getConn().NewStream(ctx, desc, method, opts...)
 }
 
+// ClientConnPoolSplitter wraps two (or more) ClientConnPools and routes
+// traffic randomly to different pools based on a client-provided traffic
+// allocation.
+type ClientConnPoolSplitter struct {
+	pools       [100]*ClientConnPool
```

Comment:
Alternatively you could have a slice of weighted pools `[]*weightedPool` where each entry and its associated percentage. Then to pick a pool, you pick a random number as you do now and keep walking until the random number falls within the allocated range. If you do that you wouldn't need a separate `uniquePools` slice.

Reply:
Yeah, I considered that but thought it was better to keep the implementation a bit simpler at the expense of a few hundred bytes of memory for this case where only full percentage increments are needed. Added a TODO with your suggestion, though.